[SaferC++] Fix more memory safety issues in WebInspectorUIExtensionController.cpp

zakariaridouh opened the pull request:
#### ed0f8b4b0492fd2bd22e4ce26edac8d3d9dd88bb
<pre>
[SaferC++] Fix more memory safety issues in WebInspectorUIExtensionController.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=300454">https://bugs.webkit.org/show_bug.cgi?id=300454</a>
<a href="https://rdar.apple.com/162303836">rdar://162303836</a>

Reviewed by Ryosuke Niwa.

Fix memory safety issues in WebInspectorUIExtensionController.cpp by
applying the SaferC++ WebKit guidelines.

* Source/WebCore/inspector/InspectorFrontendClient.h:
(WebCore::InspectorFrontendClient::protectedFrontendAPIDispatcher):
* Source/WebKit/WebProcess/Inspector/WebInspectorUIExtensionController.cpp:
(WebKit::WebInspectorUIExtensionController::parseExtensionErrorFromEvaluationResult const):
(WebKit::WebInspectorUIExtensionController::registerExtension):
(WebKit::WebInspectorUIExtensionController::unregisterExtension):
(WebKit::WebInspectorUIExtensionController::createTabForExtension):
(WebKit::WebInspectorUIExtensionController::evaluateScriptForExtension):
(WebKit::WebInspectorUIExtensionController::reloadForExtension):
(WebKit::WebInspectorUIExtensionController::showExtensionTab):
(WebKit::WebInspectorUIExtensionController::navigateTabForExtension):
(WebKit::WebInspectorUIExtensionController::evaluateScriptInExtensionTab):
(WebKit::WebInspectorUIExtensionController::didShowExtensionTab):
(WebKit::WebInspectorUIExtensionController::didHideExtensionTab):
(WebKit::WebInspectorUIExtensionController::didNavigateExtensionTab):
(WebKit::WebInspectorUIExtensionController::inspectedPageDidNavigate):

Canonical link: <a href="https://commits.webkit.org/301517@main">https://commits.webkit.org/301517@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/273f367c6482eab00d58e198ced3b2e49564742b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126202 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45860 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36674 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132986 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77984 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/971dad2d-7802-429e-bb04-dee3b731b2f6) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128073 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46529 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54408 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/96091 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/64201 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7a3ea1bf-5fe7-43b6-a0f9-4ce3e6133a5a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129150 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/37228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/112886 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/76573 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/68d9ab80-24d8-4bae-bce4-13e20a91256a) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/36128 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/31065 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76460 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/107010 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31293 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/135689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52956 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40676 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/135689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53418 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109105 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/135689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26595 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49725 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/28061 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50340 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52856 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58689 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52174 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55520 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53889 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebKit/WebProcess/Inspector/WebInspectorUIExtensionController.cpp:
```diff
@@ -76,7 +76,7 @@ std::optional<Inspector::ExtensionError> WebInspectorUIExtensionController::pars
 
     ASSERT(m_frontendClient);
-    auto globalObject = m_frontendClient->frontendAPIDispatcher().frontendGlobalObject();
+    auto globalObject = Ref { m_frontendClient->frontendAPIDispatcher() }->frontendGlobalObject();
```
Can we add `protectedFrontendAPIDispatcher()` instead of adding `Ref { ~ }` multiple times in this file?

zakariaridouh replied:
Sure.

zakariaridouh replied:
Fixed in latest version of the patch.

rniwa reviewed (approved):
Looks much nicer!

zakariaridouh committed:
[SaferC++] Fix more memory safety issues in WebInspectorUIExtensionController.cpp
https://bugs.webkit.org/show_bug.cgi?id=300454
rdar://162303836

Reviewed by Ryosuke Niwa.

Fix memory safety issues in WebInspectorUIExtensionController.cpp by
applying the SaferC++ WebKit guidelines.

* Source/WebCore/inspector/InspectorFrontendClient.h:
(WebCore::InspectorFrontendClient::protectedFrontendAPIDispatcher):
* Source/WebKit/WebProcess/Inspector/WebInspectorUIExtensionController.cpp:
(WebKit::WebInspectorUIExtensionController::parseExtensionErrorFromEvaluationResult const):
(WebKit::WebInspectorUIExtensionController::registerExtension):
(WebKit::WebInspectorUIExtensionController::unregisterExtension):
(WebKit::WebInspectorUIExtensionController::createTabForExtension):
(WebKit::WebInspectorUIExtensionController::evaluateScriptForExtension):
(WebKit::WebInspectorUIExtensionController::reloadForExtension):
(WebKit::WebInspectorUIExtensionController::showExtensionTab):
(WebKit::WebInspectorUIExtensionController::navigateTabForExtension):
(WebKit::WebInspectorUIExtensionController::evaluateScriptInExtensionTab):
(WebKit::WebInspectorUIExtensionController::didShowExtensionTab):
(WebKit::WebInspectorUIExtensionController::didHideExtensionTab):
(WebKit::WebInspectorUIExtensionController::didNavigateExtensionTab):
(WebKit::WebInspectorUIExtensionController::inspectedPageDidNavigate):

Canonical link: https://commits.webkit.org/301517@main
